### PR TITLE
[Backport 2.8] Rest admin permissions (#2411)

### PR DESCRIPTION
**Pull request by DarshitChanpura**
Backports #2411 to 2.8

### Check List
- [x] New functionality includes testing
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by willyborankin** [comment]
Sorry I forget to backport this PR https://github.com/opensearch-project/security/pull/2605

**Comment by willyborankin** [comment]
tests passed for at least Linux build

**Comment by cwperks** [comment]
@opensearch-project/engineering-effectiveness We are having to remove a dependency on backports to 2.8 in order to get our CI checks working, but the underlying issue is that there is a stale minimum distribution for OpenSearch core (https://artifacts.opensearch.org/snapshots/core/opensearch/2.8.0-SNAPSHOT/opensearch-min-2.8.0-SNAPSHOT-linux-x64-latest.tar.gz) that does not reflect the code that is on the 2.8 branch.

Can someone help produce a new minimum distribution from the 2.8 branch of core?

**Review by cwperks — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -289,7 +289,6 @@ configurations.all {
 }
 
 dependencies {
-    implementation 'jakarta.annotation:jakarta.annotation-api:1.3.5'
```
Comment (cwperks) [inline build.gradle:292]: This will be re-introduced in another PR once a new minimum snapshot distribution of core is available.

**Pull request merged**